roles/matrix-bridge-appservice-discord/defaults/main.yml

```diff
@@ -41,6 +41,23 @@ matrix_appservice_discord_bridge_homeserverUrl: "http://matrix-synapse:8008"
 matrix_appservice_discord_bridge_disablePresence: false
 matrix_appservice_discord_bridge_enableSelfServiceBridging: false
 
+# Database-related configuration fields.
+#
+# To use SQLite, stick to these defaults.
+#
+# To use Postgres:
+# - change the engine (`matrix_appservice_discord_database_engine: 'postgres'`)
+# - adjust your database credentials
+matrix_appservice_discord_database_engine: 'sqlite'
+matrix_appservice_discord_database_filename: "/data/{{ matrix_appservice_discord_database_filename_name }}"
+matrix_appservice_discord_database_filename_name: "discord.db"
+matrix_appservice_discord_database_connString: 'postgresql://{{ matrix_appservice_discord_database_connString_username }}:{{ matrix_appservice_discord_database_connString_password }}@{{ matrix_appservice_discord_database_connString_hostname }}:{{ matrix_appservice_discord_database_connString_port }}/{{ matrix_appservice_discord_database_connString_db_name }}'
+matrix_appservice_discord_database_connString_username: 'matrix_appservice_discord'
```

Comment:
I'd argue to drop the `_connString_` part of all `$role_database_*` variables, as it's not the username/password/... _of_ the connString, the connString is a compound of the set. Also, i'm lazy and dont want to type that much and shorter variable names are better, and `matrix_appservice_discord_database_username` is still descriptive.

Reply:
True!

I've gone for prefixing them with `_connString_` because:

- we already use a similar pattern in other places

Example:

```yaml
matrix_synapse_docker_image: "{{ matrix_synapse_docker_image_name_prefix }}matrixdotorg/synapse:{{ matrix_synapse_docker_image_tag }}"
matrix_synapse_docker_image_name_prefix: "{{ 'localhost/' if matrix_synapse_container_image_self_build else 'docker.io/' }}"
# The if statement below may look silly at times (leading to the same version being returned),
# but ARM-compatible container images are only released 1-7 hours after a release,
# so we may often be on different versions for different architectures when new Synapse releases come out.
matrix_synapse_docker_image_tag: "{{ 'v1.24.0' if matrix_architecture == 'amd64' else 'v1.24.0' }}"
matrix_synapse_docker_image_force_pull: "{{ matrix_synapse_docker_image.endswith(':latest') }}"
```

.. but I could see how having these variables being more standalone makes sense too (`matrix_synapse_docker_image_tag` -> `matrix_synapse_version`)

- for bridges we aim for sensible mapping between variable names and configuration values

For example `matrix_appservice_discord_database_connString` *likely* mapping to the `database.connString` configuration option of the Appservice Discord bridge.

I think for some role we even had the prefix be `(matrix_service_config_STUFF_HERE`) making it even more explicit.

`matrix_appservice_discord_database_username` somewhere near `matrix_appservice_discord_database_connString` may make people believe there's actually also a `database.username` field in the configuration.

Putting the username under the `matrix_appservice_discord_database_connString` variable prefix and having it used inside probably makes it clear enough that it's just some way to split the `connString` into sub-parts for easier definition/redefinition.

----------

You're right though.. For people who override these configuration values, putting something like `matrix_appservice_discord_database_connString_username` in their `vars.yml` file sounds weird. We should probably change to what you're proposing.

Reply:
I've put the variables without the infix `_connString_` in the `group_vars/matrix_server` for now, as i'm for hiding implementation details (such as specific config.yaml layout) from the user where it would have little benefit.

Reply:
Alright! Let's get rid of that `_conString_` infix thing. Or should I do it for Appservice Discord?

Reply:
I've redone these variable names in d91aa5a060ab0. Let me know what you think!

I've even gone for `matrix_appservice_discord_postgres_username`, instead of `matrix_appservice_discord_database_username` (`_postgres_` instead of `_database_`), to make it even more explicit what it's about.

Reply:
i am not sure if `_postgres_` is better then `_database_` - the connString format supports a lot of databases, and they all have the require information like  `host`, `port`, `user`, ... - the generic `_database_` is probably more accurate, as with that, a user could run a mysql-DB aswell and use the variables and just changeout the protocol-prefix in the connString and change the `*_database_engine` to `mysql`

Reply:
Haha.. Good point! Switching from `_postgres_` back to `_database_` does make some sense.

The protocol schema thing can be its own variable too (e.g. `matrix_appservice_discord_database_protocol_schema`).

Reply:
we _could_ extract the schema from the `*_database_engine`, but for now, i think it's fine w/ just `*_database_engine` and building the connString with a static scheme (postgres)

Reply:
I've restored naming from `_postgres_` to `_database_` in 087dbe4ddc80ba. I guess this discussion can be resolved?